Misc/NEWS.d/next/Library/2022-04-12-19-08-13.gh-issue-91404.zjqYHo.rst

```diff
@@ -0,0 +1,3 @@
+Improve the performance of :mod:`re` matching by using computed gotos (or
```

Comment:
Add that it speeds up matching by 10-20%.

Reply:
Please don't. These claimed speedups are highly misleading.

The release notes for 3.9 and 3.10 have various claimed large speedups, yet 3.9 is no faster than 3.8 and 3.10 only a little bit faster.

Reply:
The benchmark results are convincing to me.

Without this it is not clear why bother with making this change at all.

Reply:
You'll only see a 10-20% speedup for these specific benchmarks, which are contrived. 
Real programs, even those that spend a lot regexes will spend a much lower proportion of their time in the regex library.

Any number we give will be misleading.

Reply:
This branch vs 3.10:

```
- regex_v8: 147 ms +- 12 ms -> 132 ms +- 11 ms: 1.11x faster
- regex_effbot: 19.1 ms +- 1.2 ms -> 17.5 ms +- 0.9 ms: 1.10x faster
- regex_dna: 1.30 sec +- 0.01 sec -> 1.24 sec +- 0.01 sec: 1.04x faster
```

(Those numbers are very close to what you get by combining the [latest `re` perf regressions](https://github.com/faster-cpython/ideas/blob/0bc571a3c07a52eaae772af72ab0d0097d995505/main-vs-310.rst) with the numbers from this PR.) 

I'll just leave the NEWS entry as-is (very few people read it anyways). I don't have a strong opinion on what should go in What's New (but @pablogsal might).

Reply:
Perhaps the wording "up to 10% faster on the [`pyperformance` regular expression benchmarks](https://pyperformance.readthedocs.io/benchmarks.html#regex-dna)" (or similar) might suffice.

Reply:
I think this should indeed go in the What's new. Could you add a small sentence in the Performance section, maybe?